Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -133,34 +141,63 @@ internal void ValidateSecurePassword(SecureString password)
         private string ExtractPassword(SecureString password) =>
             password == null ? string.Empty : SecureStringHelper.Decode(password);
 
-        internal SFSession GetSession(string connStr, SecureString password)
+        internal SFSession GetSession(string connStr, SecureString password, SecureString passcode)
         {
             s_logger.Debug("SessionPool::GetSession" + PoolIdentification());
+            SFSession session = null;
+            var sessionProperties = SFSessionProperties.ParseConnectionString(connStr, password);
+            ValidatePoolingIfPasscodeProvided(sessionProperties);
```

Comment:
it doesn't handle the case when the passcode is provided via secure string

Reply:
As we review previously the passcode when is provided as parameter with a secure string, it only is use to create the first session, for the others sessions in the pool it should not be sent because it could only be use one time.

Reply:
What I understand is that it is reasonable to provide a passcode only when either it is username_password_mfa authentication or MinPoolSize is disabled.
So imagine someone has MinPoolSize enabled - eg. MinPoolSize = 5, and uses basic authentication.
If he would provide passcode in connection string - he would get an exception because it doesn't make sense to use the same passcode in many connections.
If he would provide passcode by secure string - he would not get any exception, but the passcode would be used also for many session creations because MinPoolSize > 0 and logic which you mentioned is only for MFA authentication.
So why not to throw an exception in such a case?

Reply:
When we pass the passcode using the secure string, only the first connection will use this value; if we ensure MinPoolSize, the other connections will not use it. If basic authentication and the user has MFA configured, an exception will be thrown by the server.

Reply:
In case of basic authentication:
- there would be created background threads each of them having the same passcode as the main thread
- each of the threads would use the same passcode and most of them (except 1) would have a failure

I know that the error would occur anyway in the communication with the server, but there exists a principle of "fail fast" If we could recognise that such a configuration leads to server errors why not to fail fast in the validation? Thanks the server would not get any requests from us for which we know that the result would be a failure. We are saving sever resources if we fail fast.